src/main/resources/hudson/plugins/ircbot/IrcPublisher/global.jelly

```diff
@@ -25,16 +25,16 @@
     <f:entry title="Channels"
         description="Channels to join. Name and optional password. Check 'Notification only' if you want to disallow bot commands."
         help="/plugin/ircbot/help-globalConfigChannels.html">
-     <f:repeatable name="irc_publisher.channels" var="ch" items="${descriptor.defaultTargets}">
-        <div width="100%">
+     <f:repeatable var="ch" field="defaultTargets">
+        <div>
             <f:entry title="Name">
-                <input type="text" width="100%" style="text-align:left" name="irc_publisher.channel.name" value="${ch.name}" />
+              <f:textbox field="name" />
             </f:entry>
             <f:entry title="Password">
-                <input type="password" width="100%" style="text-align:left" name="irc_publisher.channel.password" value="${ch.password}" />
+              <f:password field="password"/>
```

Comment:
By the way, is this the trick needed to solve https://www.jenkins.io/security/advisory/2019-04-03/#SECURITY-829 ? Does this jelly field ensure storing encrypted passwords on disk (and decrypting to use them)?

Reply:
More work is needed but yes it’s one step